### ISSUE TITLE
Assembly 'lines' method does not handle !BLOCK_CONTIGUOUS_P(block) case

### ISSUE THREAD

**Issue by Cebtenzzre** [cyrus-and/gdb-dashboard#135]
I've run into a function in a C++ program (compiled with at least the flags `-O3 -fno-plt -Wl,-O1,--sort-common,--as-needed -ggdb3`) that _appears_ to not be contiguous.  Therefore, the output of `disassemble` prints multiple address ranges:
```
Address range 0x55555567ad60 to 0x55555567b91d:
=> 0x000055555567ad60 <+0>:	push   %rbp
   0x000055555567ad61 <+1>:	mov    %rsp,%rbp
(...)
Address range 0x55555558d144 to 0x55555558d16e:
   0x000055555558d144 <+-973852>:	mov    %r12,%rdi
   0x000055555558d147 <+-973849>:	vzeroupper
(...)
```

This code in gdb's `cli-cmds.c` is responsible:
```c
if (block == nullptr || BLOCK_CONTIGUOUS_P (block))
  {
// ...
  }
else
  {
    for (int i = 0; i < BLOCK_NRANGES (block); i++)
      {
        CORE_ADDR range_low = BLOCK_RANGE_START (block, i);
        CORE_ADDR range_high = BLOCK_RANGE_END (block, i);
        printf_filtered (_("Address range %s to %s:\n"),
// ...
```

`Assembly`'s `lines` method does not expect this, so this error is printed:
```
Cannot write the dashboard
Traceback (most recent call last):
  File "<string>", line 359, in render
  File "<string>", line 917, in lines
ValueError: invalid literal for int() with base 16: 'ress'
```

This code gets the word "Address" minus the first 3 letters, which is not a base-16 integer:
```python
start = int(re.split('[ :]', output[1][3:], 1)[0], 16)
```

I've currently worked around this by catching ValueError in the try block surrounding that line.

**Comment by Cebtenzzre** [comment]
No, sorry.  I've been unable to reproduce this, as I've recompiled that executable several times since then, and I don't remember which function triggered that.  I don't know to create a "split function" like this in C either.

**Comment by cyrus-and** [comment]
It should be fixed in 45645316b1487df73d8209bf9134af1862460375, reopen if it's not.
